### ISSUE TITLE
Script no longer works, pacman 4.2 toolset removed --asroot from makepkg

### ISSUE THREAD

**Issue by BrainwreckedTech** [BrainwreckedTech/aurpb#4]
The aurpb script ran makechrootpkg as root, which in turned ran makepkg --asroot.  As of pacman 4.2, makepkg no longer has --asroot and will refuse to build as root.

In its current form, simply changing makechrootpkg to run as a regular user results in "sudo: no tty present and no askpass program specified" thus making the script fail.

The script hasn't been working for months under Arch Linux and I've been unable to attend to the issue properly, so I'm putting this one out there if someone else would like to help me tackle it.

**Comment by mibbio** [comment]
Seems that `makechrootpkg` can be executed as root user without problems. `makechrootpkg` itself passes a user defined by `builduser_uid=${SUDO_UID:-$UID}` to `makepkg`, which is either the current non-root user or the user who started the script by `sudo`.

**Comment by BrainwreckedTech** [comment]
Thank for pointing me in the right direction.

When I first made this script I used sudo where needed.  The change from using makepkg to makechrootpkg facilitated the need to use su, which in turn facilitated inversing the normal philosophy of "only run escalated when needed" to "run escalated but de-escalate when needed."  Didn't like it, but it made it work.

With the pacman 4.2 update, that no longer works.

Going back to the safer escalation philosophy seems to make it work.

**Comment by mibbio** [comment]
Maybe this bug report is relevant for the current `makechrootpkg` problem.
https://bugs.archlinux.org/task/43432

**Comment by BrainwreckedTech** [comment]
That's exactly what I experienced.  As long as you use `sudo` as a regular user `makechrootpkg` runs fine.  The problem is running `makechrootpkg` as root.  

And it looks like `makechrootpkg` is facing an overhaul of its own.

I have code that seems to work.

TL;DR: I can't be arsed to pour over the code like I should and I don't like it.  

Full story:  I had my back lock up the Sunday before Christmas.  My application for insurance through healthcare.gov from November _is still pending_.  My doctor doesn't want to put me on full medical leave and neither does work.  I have a narrowing of the spinal canal which explains the lack of recovering the varying degrees of partially lost feeling from the waist down.  I now have to see a specialist which I can't afford without insurance.  I have bills building up because I can't work, which for some reason I have to keep checking in with work as if I'm just playing this up and given enough pressure I'll either give up my supposed game or quit.  Applying for assistance is a joke -- by the time it would get processed I'll either have insurance and be better or out of a job I could have held on to with proper documentation.

**Comment by mibbio** [comment]
As workaround i just check if `SUDO_USER` is not emtpy and `SUDO_UID` is not 0, while the effective user is `root`. This way the script will exit with an error message if it's not invoked by a regular user directly or by using `sudo`.
You can see the changes in my [fs_43432](https://github.com/mibbio/aurpb/tree/fs_43432) branch.
I can create a pull request if needed, Hope you get better soon.
